lib/rules/no-empty.js

```diff
@@ -16,6 +16,8 @@ const astUtils = require("../util/ast-utils");
 
 module.exports = {
     meta: {
+        type: "problem",
```

Comment:
Is there a reason this is a `problem` while `no-empty-function` and `no-empty-pattern` are not?

Reply:
This was just done by mapping our current categories to the new rule types, no other reason. Seems like "suggestion" is probably the right approach for all of these.